src/Symfony/Component/Console/Input/Input.php

```diff
@@ -171,4 +191,62 @@ public function getStream()
     {
         return $this->stream;
     }
+
+    /**
+     * Returns a stringified representation of the options passed to the command.
+     * The options must NOT be escaped as otherwise passing them to a `Process` would result in them being escaped twice.
+     *
+     * @param string[] $optionNames Names of the options returned. If null, all options are returned. Requested options
+     *                              that either do not exist or were not passed (even if the option has a default value)
+     *                              will not be part of the method output.
+     *
+     * @return list<string>
+     */
+    public function unparse(?array $optionNames = null): array
+    {
+        $rawOptions = $this->getRawOptions();
+
+        $filteredRawOptions = null === $optionNames
+            ? $rawOptions
+            : array_intersect_key($rawOptions, array_fill_keys($optionNames, ''));
+
+        $unparsedOptions = [];
+
+        foreach ($filteredRawOptions as $optionName => $parsedOption) {
+            $unparsedOption = self::unparseOption(
+                $this->definition->getOption($optionName),
+                $optionName,
+                $parsedOption,
+            );
+
+            $unparsedOptions[] = \is_array($unparsedOption) ? $unparsedOption : [$unparsedOption];
```

Comment:
I think the code is simpler if you remove the private functions and you directly create arrays of strings:
```suggestion
            $option = $this->definition->getOption($optionName);
            $unparsedOptions[] = match (true) {
                $option->isNegatable() => [\sprintf('--%s%s', $value ? '' : 'no-', $name)],
                !$option->acceptValue() => [\sprintf('--%s', $name)],
                $option->isArray() => array_map(fn ($item) => sprintf('--%s=%s', $name, $item), $value),
                default => [sprintf('--%s=%s', $name, $value)],
            };
```

Reply:
I personally find it harder to understand but as you wish